editoast/editoast_authz/src/lib.rs

```diff
@@ -48,9 +61,11 @@ pub enum Error<StorageError: std::error::Error> {
     #[error("unknown subject {0}")]
     UnknownSubject(i64),
     #[error(transparent)]
-    OpenFga(#[from] fga::client::RequestFailure),
+    OpenFga(#[from] fga::client::QueryError),
     #[error(transparent)]
     Storage(StorageError),
+    #[error("Unauthorized")]
+    Unauthorized(),
```

Comment:
I'd refrain from interpreting an authorization missing as a failure *in the context of `editoast_authz`*. This crate provides the required mechanism to determine whether an operation is authorized, so a missing authorization is a valid result in this context. Interpreting this as an error is a job for the `views` imo.

If we had instead a type like:

```rust
enum Authorization<ProtectedData> {
	Authorized(ProtectedData),
	Unauthorized
}

fn retrieve_protected_data(...) -> Result<Authroized<T>
, Error>
```

it makes the authorization obvious in the signature (instead of hidden as one of many variants in a common error type) and forces the user to explicitly match on the authorization and deal with the unauthorized case.

Reply:
I still think the distinction is important, but this can be done afterwards

Reply:
This error is not used (at least for now) when we retrieve some data, but when we do some grant &revoke operations, like in `grant_infra_writer` . These functions return nothing, but throw an error if the user is not authorized to do the action, which make sense for me. 

I don't know what are the good practices in rust for managing exception, but in the java world working with exception like that is normal.

If I implement ` grant_infra_writer`  in your way,  It becomes something like that 
`pub async fn grant_infra_owner( &self,  user_id: i64,  infra_id: i64,) -> Result<Authorization<()>, Error<S::Error>> `, and then I need to check the returned result to know if the action was well performed or not.  
This is weird for me, because we already have a `Result` which can be in Success or in Error mode.

Reply:
Following our discussion, this still requires some debate. So let's just fix the form here:

```suggestion
    #[error("Unauthorized")]
    Unauthorized,
```

likewise for returning it: `Err(Error::Unauthorized)` without parentheses